### PR TITLE
Add linkdeps flag which enables passing flags to link dependent libraries

### PR DESCRIPTION
**Pull request by matthew-zitara**
This may not work with all compilers but 1. we can disable it and link manually still and 2. it should be easy to fix if compilers need different handling

### CODE REVIEW

File: foreign_cc/private/make_env_vars.bzl [hunk 1]
```diff
@@ -30,18 +31,33 @@ def get_make_env_vars(
...
+        for lib in reversed(inputs.libs):
+            libname = lib.basename
+            ext = lib.extension
+            if ext:
+                libname = libname[:-1-len(ext)]
+            if libname.startswith('lib'):
+                libname = libname[3:]
+            libname = "-l" + libname
+            if not lib_set.get(libname):
+                lib_set[libname] = 1
```
Comment (xander-zitara) [inline foreign_cc/private/make_env_vars.bzl:60]: I am guessing this is some starlark limitation that we can't write this in a more pythonic way? Like for instance
```suggestion
            if not libname in lib_set:
                lib_set.add(libname)
```

**Review by xander-zitara — APPROVED** [review]
Looks good, have a question

**Comment by xander-zitara** [comment]
> Looks good, have a question

Oh and: do you have an example of using linkdeps? Do we need to change our use of `make()`, and what would that look like?

File: foreign_cc/private/make_env_vars.bzl (hunk 1 above)
Comment (matthew-zitara) [reply foreign_cc/private/make_env_vars.bzl:60]: Apparently starlark doesn't support `set`s. https://docs.bazel.build/versions/main/skylark/language.html#differences-with-python . I kept the same style as `gen_dirs_set` in this function

**Pull request merged**